### ISSUE TITLE
item conduit voiding item

### ISSUE THREAD

**Issue by Petrisis** [SleepyTrousers/EnderIO#5456]
#### Issue Description:
An EnderIO Item Conduit attached to an Trash Chest (ExtraUtilities2) deletes 4 items like a pull into the void. It works totally fine on other types of chests. It even works with just a single item conduit attached to nothing else; see screenshot.
![RemoveFeed](https://user-images.githubusercontent.com/31795322/95795368-76cda000-0cea-11eb-8f72-a2722855f82c.png)
The conduit is set to always active, green, no round robin, no self feed, no filter. If we set the conduit to not extract or never active, the disappearing of items stops.
If connected to another vanilla chest, it pumps items out of the trash chest with 8 items/interval speed - 4 going into the chest and 4 disappearing.

#### What you expected to happen:
I expected the conduit to do nothing with no viable destination.
the purpose of the trash chest is to collect items until full. if an item cant get inserted, the chest voids it.

#### Steps to reproduce:

1. put down ExUtils2 Trash Chest
2. connect EnderIO Item conduit to either side, activate its extraction
3. put a stack of anything into the chest
4. see the items disappearing

____
#### Affected Versions:

FTB: Interactions 2.0.9
- EnderIO: 5.2.59
- EnderCore: 1.12.2-0.5.76
- Minecraft: 1.12.2
- Forge: 14.23.5.2847
- SpongeForge? no
- Optifine? no
- Single Player/**Server**?

#### Your most recent log file where the issue was present: 
https://pastebin.com/Bkk7vwHA

**Comment by CitiesXL2815** [comment]
also your pastebin not found

**Comment by HenryLoenwind** [comment]
The conduit code only extracts items from a source inventory AFTER it successfully inserted them into a target. So if items are disappearing from that chest, it must delete them when we ask it about what is inside.

About skipping everything when there's no target at all: People don't usually enable extraction unless they also add a target, but it's a simple optimisation to add as we do have a list of targets available at that point.

**Comment by Petrisis** [comment]
sry im not very familiar with pastebin. i created a new one:
https://pastebin.com/i5MMXfiG

**Comment by HenryLoenwind** [comment]
Are you sure you don't have a duping issue instead?

`[00:00:00] [Server thread/WARN] [enderio]: NetworkedInventory.itemExtracted: Inserted 1 Greatwood Log but only removed 0 Air from com.rwtema.extrautils2.tile.TileTrashChest$1@7d8c126e at BlockPos{x=-987, y=67, z=-12}. This means that 1 items were just duped by com.rwtema.extrautils2.tile.TileTrashChest$1@7d8c126e!`

That Trash Chest promises us some greatwood but when we actually want to take it that it doesn't give it to us. Here, btw, you can see clearly that we don't remove items from the source until we got a place to put them and already inserted them there.

**Comment by Ommina** [comment]
As an aside, kinda.  Because I need to get out more.

With conduits: 64 items in, 32 in destination chest.  As described.

With a vanilla hopper.  Trash Chest -> Hopper -> Vanilla Chest.  Put 64 items into the Trash Chest, got 32 delivered to the vanilla chest.

Tried with Thermal Expansion item ducts.  Put 64 in, got 64 (!) out.

Tried with (/cough) Extra Utilities 2 Transfer Node (Items).  Put 64 in, got 32 out.

**Comment by HenryLoenwind** [comment]
yes, it is clear that that chest voids items when you ask it what's in there. If you just blindly extract instead of asking first...you're sitting on an item that you need to void yourself if you don't find a place to put it.
